app/helpers/repositories_helper.rb

```diff
@@ -155,131 +166,85 @@ def to_utf8_internal(str)
 
-  def subversion_field_tags(form, repository)
-    url = content_tag('div', class: 'form--field') do
-      form.text_field(:url,
-                      size: 60,
-                      required: true,
-                      disabled: (repository && !repository.root_url.blank?)) +
-      content_tag('div',
-                  'file:///, http://, https://, svn://, svn+[tunnelscheme]://',
-                  class: 'form--field-instructions')
-    end
-
-    login = content_tag('div', class: 'form--field') do
-      form.text_field(:login, size: 30)
-    end
-
-    pwd = content_tag('div', class: 'form--field') do
-      form.password_field(:password,
-                          size: 30,
-                          name: 'ignore',
-                          value: ((repository.new_record? || repository.password.blank?) ? '' : ('x' * 15)),
-                          onfocus: "this.value=''; this.name='repository[password]';",
-                          onchange: "this.name='repository[password]';")
-    end
-
-    url + login + pwd
+  def scm_vendor_tag(repository)
+    select_tag('scm_vendor',
+               scm_options(repository),
+               class: 'form--select repositories--remote-select',
+               data: {
+                 remote: true,
+                 url: url_for(controller: '/repositories',
+                              action: 'edit', project_id: @project.id),
+               },
+               disabled: (repository && !repository.new_record?)
+              )
   end
 
-  def git_field_tags(form, repository)
-    url = content_tag('div', class: 'form--field -required') do
-      form.text_field(:url,
-                      label: :label_git_path,
-                      size: 60,
-                      disabled: (repository && !repository.root_url.blank?)) +
-      content_tag('div',
-                  l(:text_git_repo_example),
-                  class: 'form--field-instructions')
-    end
-
-    encoding = content_tag('div', class: 'form--field') do
-      form.select(:path_encoding,
-                  [nil] + Setting::ENCODINGS,
-                  label: l(:label_path_encoding)) +
-      content_tag('div',
-                  l(:text_default_encoding),
-                  class: 'form--field-instructions')
-    end
-
-    url + encoding
+  def git_path_encoding_options(repository)
+    default = repository.new_record? ? 'UTF-8' : repository.path_encoding
+    options_for_select(Setting::ENCODINGS, default)
   end
 
-  def filesystem_field_tags(form, repository)
-    url = content_tag('div', class: 'form--field -required') do
-      form.text_field(:url,
-                      label: :label_filesystem_path,
-                      size: 60,
-                      disabled: (repository && !repository.root_url.blank?))
-    end
+  ##
+  # Determines whether the repository settings save button should be shown.
+  # By default, it is not shown when repository exists and is managed.
+  def show_settings_save_button?(repository)
```

Comment:
Unused method argument - `repository`. If it's necessary, use `_` or `_repository` as an argument name to indicate that it won't be used. You can also write as `show_settings_save_button?(*)` if you want the method to accept any arguments but don't care about them.